### PR TITLE
Ganglia int

### PR DESCRIPTION
**Pull request by rportilla-databricks**
Ganglia support in DCS has been a long-standing request from customers - this PR contains the artifacts (shell scripts, monit and other ganglia dependencies) required to start Ganglia services and monitoring on a DCS cluster.

### CODE REVIEW

File: experimental/ubuntu/ganglia/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+# Runtime 5.2. Once we move to Runtime 6, we should be able to use a newer version of tornado
+# because PythonShell should not depend on nbconvert.
+RUN export GANGLIA_API=ganglia-api-cbe773d051168e05118774708ff7a0ce881617f4 \
+  && wget https://s3-us-west-2.amazonaws.com/databricks-build-files/$GANGLIA_API.tar.gz \
+  && tar xvzf $GANGLIA_API.tar.gz \
+  && pip install tornado==5.1.1 \
+  && apt-get install ganglia-monitor-python \
+  && find /etc/ganglia/conf.d/*.pyconf ! -name 'nvidia.pyconf' -type f -exec rm -f {} + \
+  && find /etc/ganglia/conf.d/*.pyconf.disabled ! -name 'nvidia.pyconf' -type f -exec rm -f {} + \
+  && find /usr/lib/ganglia/python_modules/*.py ! -name 'nvidia.py' -type f -exec rm -f {} +
```
Comment (evanye) [inline experimental/ubuntu/ganglia/Dockerfile:62]: did you test the ganglia api by curling the driver proxy? I'm not sure if there's enough usage to warrant trying to support this, and could be left out if it doesn't work

File: experimental/ubuntu/ganglia/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# Stop Ganglia related services from starting on boot
+RUN update-rc.d ganglia-monitor disable \
+  && update-rc.d gmetad disable \
+  && update-rc.d apache2 disable
+
+# create directories for unix monitoring tool required for ganglia
+# two services needed by Ganglia are gmetad and ganglia-monitor
+RUN mkdir -p /etc/monit/conf.d
+
+SHELL ["/bin/bash", "-c"]
```
Comment (evanye) [inline experimental/ubuntu/ganglia/Dockerfile:72]: this isn't needed - it'll be specified when the container is launched

File: experimental/ubuntu/ganglia/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+RUN echo $'check process gmetad with pidfile /var/run/gmetad.pid\n\
+      start program = "/usr/sbin/service gmetad start"\n\
+      stop program = "/usr/sbin/service gmetad stop"\n\
+      if memory usage > 500 MB for 3 cycles then restart\n\
+\n\
+check process apache2 with pidfile /var/run/apache2/apache2.pid\n\
+      start program = "/usr/sbin/service apache2 start"\n\
+      stop program = "/usr/sbin/service apache2 stop"\n\
+      if memory usage > 500 MB for 3 cycles then restart\n\
+' > /etc/monit/conf.d/gmetad-not-active
```
Comment (evanye) [inline experimental/ubuntu/ganglia/Dockerfile:89]: should be easier to do this via ADD or COPY docker primitives

**Review by evanye — APPROVED** [review]
Nice work! left some minor suggestions, feel free to squash and merge. Please add a sentence to the pr description

File: experimental/ubuntu/ganglia/Dockerfile (hunk 2 above)
Comment (rportilla-databricks) [reply experimental/ubuntu/ganglia/Dockerfile:72]: @evanye what exactly is not needed? All the /etc/monit/conf.d files?

Comment (evanye) [reply experimental/ubuntu/ganglia/Dockerfile:72]: Just the `SHELL ["/bin/bash", "-c"]` line

**Review by evanye — APPROVED** [review]

File: experimental/ubuntu/ganglia/Dockerfile (hunk 2 above)
Comment (rportilla-databricks) [reply experimental/ubuntu/ganglia/Dockerfile:70]: Thanks, removed this.

File: experimental/ubuntu/ganglia/Dockerfile (hunk 1 above)
Comment (rportilla-databricks) [reply experimental/ubuntu/ganglia/Dockerfile:62]: I didn't so will remove these for now. I think there will be customers who might want it, so we can add a separate example for this.

File: experimental/ubuntu/ganglia/Dockerfile (hunk 3 above)
Comment (rportilla-databricks) [reply experimental/ubuntu/ganglia/Dockerfile]: Updated to remove inline shell commands and used ADD

**Pull request merged**